### DISCUSSION TITLE
Tailwind size classes (https://tailwindcss.com/docs/size) are missing in development and production build

### DISCUSSION THREAD

**Discussion by mimo-sha256** [tailwindlabs/tailwindcss#13559] [Help] [answered]
I'm working on a NextJS project with TailwindCSS.

I am using a code snippet from Preline components:
`<div className="bg-white border rounded-xl shadow-sm sm:flex dark:bg-neutral-900 dark:border-neutral-700 dark:shadow-neutral-700/70">
        <div className="flex-shrink-0 relative w-full rounded-t-xl overflow-hidden pt-[40%] sm:rounded-s-xl sm:max-w-60 md:rounded-se-none md:max-w-xs">
          <img
            className="size-full absolute top-0 start-0 object-cover"
            src="https://images.unsplash.com/photo-1680868543815-b8666dba60f7?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2532&q=80"
            alt="Image Description"
          />
        </div>
        <div className="flex flex-wrap">
          <div className="p-4 flex flex-col h-full sm:p-7">
            <h3 className="text-lg font-bold text-gray-800 dark:text-white">
              Card title
            </h3>
            <p className="mt-1 text-gray-500 dark:text-neutral-400">
              Some quick example text to build on the card title and make up the
              bulk of the card's content.
            </p>
            <div className="mt-5 sm:mt-auto">
              <p className="text-xs text-gray-500 dark:text-neutral-500">
                Last updated 5 mins ago
              </p>
            </div>
          </div>
        </div>
      </div>`

The `size-full` class is not loaded and the style is missing from layout.css.
If I add other classes (for example `hue-rotate-15`) it does appear, but the size classes do not.

Tailwing config
`/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    "./src/pages/**/*.{js,ts,jsx,tsx,mdx}",
    "./src/components/**/*.{js,ts,jsx,tsx,mdx}",
    "./src/app/**/*.{js,ts,jsx,tsx,mdx}",
  ],
  theme: {
    extend: {
      backgroundImage: {
        "gradient-radial": "radial-gradient(var(--tw-gradient-stops))",
        "gradient-conic":
          "conic-gradient(from 180deg at 50% 50%, var(--tw-gradient-stops))",
      },
      spacing: {
        128: "32rem",
      },
    },
  },
  plugins: [],
  safelist: ["size-full"],
};`

I tried adding `size-full` under safelist but it still does not work.

Postcss config
`module.exports = {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
  },
}
`

I do not have purge css installed either.

What could be the issue here?

**Comment by wongjn** [comment, marked as answer]
It could be that you are running an older version of Tailwind. The `size-*` utilities were introduced in `v3.4.0`.

**Reply by mimo-sha256** [reply]
That was it, thank you! I did an npm update thinking it would get the latest version but did not cross check the version numbers.
